python-threatexchange/threatexchange/signal_type/index.py

```diff
@@ -48,6 +48,19 @@ def __init__(self, distance: int, metadata: T) -> None:
         self.metadata = metadata
 
 
+class VPDQIndexMatch(IndexMatch):
```

Comment:
You have to pass through T here to get it type 
```suggestion
class VPDQIndexMatch(IndexMatch[T]):
```